sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs

```diff
@@ -1009,6 +1041,90 @@ public async Task GetBatchSettingIfChangedWithUnmodifiedPageSync()
             Assert.AreEqual(2, pagesCount);
         }
 
+        [RecordedTest]
+        [AsyncOnly]
+        [ServiceVersion(Min = ConfigurationClientOptions.ServiceVersion.V2023_10_01)]
+        public async Task GetBatchSettingIfChangedWithUnmodifiedPageDoesNotLogWarningAsync()
```

Comment:
Adding tests to the new `AsPages` extension method as well since it could be subject to the same bug if we eventually change the code where 304 responses are handled.